app/questionnaire/questionnaire_schema.py

```diff
@@ -728,3 +733,50 @@ def _get_error_messages(self) -> dict:
             messages.update(self.json["messages"])
 
         return messages
+
+    def _populate_when_rules_section_dependencies(self) -> None:
+        for section in self.get_sections():
+            when_rules = self._get_values_for_key(section, "when")
+            rules: Union[Mapping, list] = next(when_rules, [])
+
+            if rules_section_dependencies := self._get_rules_section_dependencies(
+                section["id"], rules
+            ):
+                self._when_rules_section_dependencies_map[
+                    section["id"]
+                ] = rules_section_dependencies
+
+    def _get_rules_section_dependencies(
+        self, current_section_id: str, rules: Union[Mapping, Sequence]
+    ) -> set[str]:
+        rules_section_dependencies: set[str] = set()
+
+        if isinstance(rules, dict) and any(
+            operator in rules for operator in OPERATION_MAPPING
+        ):
+            rules = self.get_operands(rules)
+
+        for rule in rules:
+            if not isinstance(rule, dict):
+                continue
+
+            answer_id = None
+
+            if "id" in rule:
+                answer_id = rule["id"]
```

Comment:
```suggestion
            answer_id = rule.get("id")
```

Reply:
Because of the elif, I am not convinced that would add anything, I would then need to something like  
```            
if not answer_id and rule.get("source") == "answers":
      answer_id = rule.get("identifier")
```